shimming-toolbox/shimmingtoolbox/load_nifti.py

```diff
@@ -210,3 +210,103 @@ def read_nii(fname_nifti, auto_scale=True):
+
+
+def get_isocenter(json_data):
+    """ Get the isocenter location in RAS coordinates from the json file. The patient position is used to infer the
+    table position in the patient coordinate system. When the table is at (0,0,0), the origin is at the isocenter.
+    We can therefore infer the isocenter as -table_position when the table_position is in RAS coordinates.
+
+    Args:
+        json_data (dict): Dictionary containing the BIDS sidecar information
+
+    Returns:
+        numpy.ndarray: Isocenter location in RAS coordinates
+    """
+    table_position = json_data.get('TablePosition')
+    if table_position is None:
+        raise ValueError("Table position not found in json sidecar.")
+
+    patient_position = json_data.get('PatientPosition')
+    if patient_position is None:
+        raise ValueError("Patient position not found in json sidecar.")
+
+    # From the Bid specification "TablePosition":
+    # The table position, relative to an implementation-specific reference point, often the isocenter. Values must be
+    # an array (1x3) of three distances in millimeters in absolute coordinates (world coordinates). If an observer
+    # stands in front of the scanner looking at it, a table moving to the left, up or into the scanner (from the
+    # observer's point of view) will increase the 1st, 2nd and 3rd value in the array respectively. The origin is
+    # defined by the image affine.
+    table_position = np.array(table_position)
+
+    # Convert table position to RAS coordinates
+    table_position_ras = np.zeros(3)
+    if patient_position == 'HFS':
+        table_position_ras = table_position
+    elif patient_position == 'HFP':
+        table_position_ras[0] = -table_position[0]
+        table_position_ras[1] = -table_position[1]
+        table_position_ras[2] = table_position[2]
+    elif patient_position == 'FFS':
+        table_position_ras[0] = -table_position[0]
+        table_position_ras[1] = table_position[1]
+        table_position_ras[2] = -table_position[2]
+    elif patient_position == 'FFP':
+        table_position_ras[0] = table_position[0]
+        table_position_ras[1] = -table_position[1]
+        table_position_ras[2] = -table_position[2]
+    elif patient_position == 'LFP':
+        table_position_ras[2] = -table_position[0]
+        table_position_ras[1] = -table_position[1]
+        table_position_ras[0] = -table_position[2]
+    elif patient_position == 'LFS':
+        table_position_ras[2] = table_position[0]
+        table_position_ras[1] = table_position[1]
+        table_position_ras[0] = -table_position[2]
+    elif patient_position == 'RFP':
+        table_position_ras[2] = table_position[0]
+        table_position_ras[1] = -table_position[1]
+        table_position_ras[0] = table_position[2]
+    elif patient_position == 'RFS':
+        table_position_ras[2] = -table_position[0]
+        table_position_ras[1] = table_position[1]
+        table_position_ras[0] = table_position[2]
```

Comment:
I think it would be more intuitive if `table_position_ras[]` would keep the same order rather than `table_position[]` as it is the input

Reply:
fixed in [9cec368](https://github.com/shimming-toolbox/shimming-toolbox/pull/539/commits/9cec3681745602df2ee4c9edfba0108c9ede531d)

Reply:
Fixed in [9cec368](https://github.com/shimming-toolbox/shimming-toolbox/pull/539/commits/9cec3681745602df2ee4c9edfba0108c9ede531d)